plugins/google-cloud-pubsub/include/saver.hpp

```diff
@@ -0,0 +1,91 @@
+#pragma once
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2024 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <tenzir/argument_parser.hpp>
+#include <tenzir/argument_parser2.hpp>
+#include <tenzir/plugin.hpp>
+
+#include <arrow/filesystem/azurefs.h>
+#include <arrow/filesystem/filesystem.h>
+#include <arrow/filesystem/type_fwd.h>
+#include <arrow/io/api.h>
+#include <arrow/util/uri.h>
+#include <google/cloud/pubsub/publisher.h>
+
+namespace tenzir::plugins::google_cloud_pubsub {
+
+namespace pubsub = ::google::cloud::pubsub;
+
+class saver final : public plugin_saver {
+public:
+  struct args {
+    located<std::string> project_id;
+    located<std::string> topic_id;
+
+    auto add_to(argument_parser& parser) -> void {
```

Comment:
This is a header file with several non-inline function definitions. I think you should either move them out or make them inline.

Reply:
They are all defined in-class, so they are implicitly inline definitions. I can add an explicit `inline` specifier though.